metron-interface/metron-rest/src/main/java/org/apache/metron/rest/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,89 @@
+ * limitations under the License.
+ */
+package org.apache.metron.rest.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.logout.HttpStatusReturningLogoutSuccessHandler;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.sql.DataSource;
+import java.util.Arrays;
+
+@Configuration
+@EnableWebSecurity
+@Controller
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    private static final String CSRF_ENABLE_PROFILE = "csrf-enable";
+
+    @Autowired
+    private Environment environment;
+
+    @RequestMapping({"/login", "/logout", "/sensors", "/sensors*/**"})
+    public String handleNGRequests() {
+        return "forward:/index.html";
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/", "/home", "/login").permitAll()
+                .antMatchers("/app/**").permitAll()
+                .antMatchers("/vendor/**").permitAll()
+                .antMatchers("/fonts/**").permitAll()
+                .antMatchers("/assets/images/**").permitAll()
+                .antMatchers("/*.js").permitAll()
+                .antMatchers("/*.ttf").permitAll()
+                .antMatchers("/*.woff2").permitAll()
+                .anyRequest().authenticated()
+                .and().httpBasic()
+                .and()
+                .logout()
+                .logoutSuccessHandler(new HttpStatusReturningLogoutSuccessHandler())
+                .invalidateHttpSession(true)
+                .deleteCookies("JSESSIONID");
+        if (Arrays.asList(environment.getActiveProfiles()).contains(CSRF_ENABLE_PROFILE)) {
+            http.csrf().csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse());
+        } else {
+            http.csrf().disable();
+        }
+    }
+
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    public void configureJdbc(AuthenticationManagerBuilder auth) throws Exception {
+        auth
+                .jdbcAuthentication()
+                .dataSource(dataSource)
+                .withUser("user").password("password").roles("USER").and()
+                .withUser("user1").password("password").roles("USER").and()
```

Comment:
I forget exactly how spring handles these. If I recall, I believe they are always going to be there. Correct? If so we should probably only add these if we are using a specific profile (eg. only do this for dev).